src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -187,96 +190,110 @@ void ReconstructionEngine_sequentialSfM::RobustResectionOfImages(
     // and perform a BA for each camera because it makes the results
     // more stable and it's quite cheap because we have few data.
     static const std::size_t nbFirstUnstableCameras = 30;
-
+    
     if (_sfm_data.GetPoses().size() < nbFirstUnstableCameras)
     {
       // Add images one by one to reconstruct the first cameras.
       ALICEVISION_LOG_DEBUG("RobustResectionOfImages : beginning of the incremental SfM" << std::endl
                         << "Only the first image of the resection group is used." << std::endl
                         << "First image ViewId : " << vec_possible_resection_indexes.front() << std::endl
                         << "# unstable poses : " << _sfm_data.GetPoses().size() << " / " << nbFirstUnstableCameras << std::endl);
-
+      
       vec_possible_resection_indexes.resize(1);
     }
-
+  
     ALICEVISION_LOG_DEBUG("Resection group start " << resectionGroupIndex << " with " << vec_possible_resection_indexes.size() << " images.\n");
     auto chrono_start = std::chrono::steady_clock::now();
     bool bImageAdded = false;
-
+    
     // get reconstructed views before resection
     const std::set<IndexT> prevReconstructedViews = _sfm_data.getValidViews();
-
+    
     // add images to the 3D reconstruction
-    for (const size_t possible_resection_index: vec_possible_resection_indexes )
+#pragma omp parallel for 
+    for (std::size_t i = 0; i < vec_possible_resection_indexes.size(); i++)
     {
-      const size_t currentIndex = imageIndex;
-      ++imageIndex;
-
+      // Limit to a maximum number of cameras added to ensure that
+      // we don't add too much data in one step without bundle adjustment.
+      if(_sfm_data.getValidViews().size() - prevReconstructedViews.size() <= maxImagesPerGroup)
```

Comment:
Should be done before the loop, like:
```
// Limit to a maximum number of cameras added to ensure that
// we don't add too much data in one step without bundle adjustment.
if(vec_possible_resection_indexes.size() > maxImagesPerGroup)
  vec_possible_resection_indexes.resize(maxImagesPerGroup);
```

It will not be exactly the same than before, as the limit was on the number of successful resections but that makes more sense for parallelization.